docs/testing.md

```diff
@@ -0,0 +1,179 @@
+>
+> - Uses an external system (such as the thermometer in the mocking paragraph)
+> - Does a network call to other application
+> - Does any kind of file or database I/O
+>
+> Then it falls into the “integration test” category.
+
+## Component Tests
+
+React components are responsible for rendering your app, and users will directly interact with their output. Even if you have high unit and integration testing coverage and your app logic is correct, without component tests you may still deliver a broken UI to your users.
+
+For testing React components, there are two things you may want to test:
+
+- Interaction: to ensure the component behaves correctly when interacted with by a user (eg. when user presses a button)
+- Rendering: to ensure the component render output used by React is correct
+
+For example, if you have a button that has an `onPress` listener, you want to test that the button both appears correctly and that tapping the button is correctly handled by the component.
+
+There are several libraries that can help you testing these:
+
+- React’s [test renderer](https://reactjs.org/docs/test-renderer.html) is a small library provided by Facebook useful for snapshot testing. It “renders” your component view hierarchy to pure JavaScript objects held in memory. You can then make assertions on those objects in your Jest tests.
+- [`react-native-testing-library`](https://github.com/callstack/react-native-testing-library) builds on top of React’s test renderer and adds `fireEvent` and `query` apis described in the next paragraph.
+- [`@testing-library/react-native`](https://www.native-testing-library.com/) is another alternative that also builds on top of React’s test renderer and adds `fireEvent` and `query` apis described in the next paragraph.
+
+> Note that component tests are only JavaScript tests running on Node, they do _not_ take into account any iOS or Android code which is backing the React Native components. It follows that they cannot give you a 100% confidence that everything works for the user - if there is a bug in the iOS or Android code, they will not find it.
+
+### Testing user interactions
+
+Aside from rendering some UI, your components handle events like `onChangeText` for `TextInput` or `onPress` for `Button`. They may also contain other functions and event callbacks. Consider the following example:
+
+```js
+class GroceryShoppingListCreator extends React.Component {
+  render() {
+    <TextInput testID="groceryTextInput" onChangeText={this.setGroceryItem} />;
+    <Button
+      label="Add the item to list"
+      onPress={this.addNewItemToShoopingList}
+    />;
+  }
+
+  addNewItemToShoppingList = (someGroceryItem: string) => {
+    // ...
+  };
+}
+```
+
+When testing user interactions, test the component from the user perspective: you could write your tests to call `addNewItemToShoppingList` directly, and while such test works, it’s very close to the implementation details of your component.
```

Comment:
s/it’s very close to the implementation details of your component/it would break if you refactor the implementation details of your component/

Reply:
I reworded that to 

> and while such test works, it's very close to the implementation details of your component and would break if you refactor the component.

hope that works